tests/e2e.rs

```diff
@@ -200,3 +201,70 @@ async fn e2e_punish_publication_of_revoked_commit_transaction() {
         .await
         .unwrap();
 }
+
+#[tokio::test]
+async fn e2e_channel_collaborative_close() {
+    let tc_client = testcontainers::clients::Cli::default();
+    let bitcoind = Bitcoind::new(&tc_client, "0.19.1").unwrap();
+
+    let time_lock = 1;
+    let (initial_alice_balance, initial_bob_balance) = (Amount::ONE_BTC, Amount::ONE_BTC);
+
+    let create::Init {
+        alice,
+        alice_wallet,
+        bob,
+        bob_wallet,
+    } = create::Init::new(
+        &bitcoind,
+        initial_alice_balance,
+        initial_bob_balance,
+        time_lock,
+    )
+    .await;
+
+    let create::Final { alice, bob } = create::run(&alice_wallet, alice, &bob_wallet, bob).await;
+
+    alice_wallet
+        .0
+        .send_raw_transaction(alice.signed_TX_f.clone())
+        .await
+        .unwrap();
+
+    let close::Init { alice, bob } = close::Init::new(alice, bob);
+
+    let alice_final_address = alice_wallet.0.new_address().await.unwrap();
+    let bob_final_address = bob_wallet.0.new_address().await.unwrap();
+
+    let close::Final { alice, bob } =
+        close::run(alice, alice_final_address, bob, bob_final_address).unwrap();
+
+    let alice_closing_transaction = alice.into_transaction();
+    let bob_closing_transaction = bob.into_transaction();
+    assert_eq!(alice_closing_transaction, bob_closing_transaction);
+
+    // ugly wait for 1 block
+    tokio::time::delay_for(std::time::Duration::from_millis(2_000)).await;
+    let before_closure_amount = alice_wallet.0.balance().await.unwrap();
+
+    alice_wallet
+        .0
+        .send_raw_transaction(alice_closing_transaction)
+        .await
+        .unwrap();
+
+    // ugly wait for 1 block
+    tokio::time::delay_for(std::time::Duration::from_millis(2_000)).await;
+    let after_closure_amount = alice_wallet.0.balance().await.unwrap();
+
+    // amounts should not be the same as after closure should be bigger than before
+    // closure
+    assert_eq!(before_closure_amount < after_closure_amount, true);
+
+    // difference should be last channel balance
+    let amount = after_closure_amount - before_closure_amount;
+    assert_eq!(
+        amount <= initial_alice_balance && amount > Amount::ZERO,
```

Comment:
:question: Isn't it just `final_amount == latest_balance - closing_fee` (in this case `10_000`)?

Reply:
I think you just need this assertion. You should check for Bob too though.

Reply:
> ❓ Isn't it just final_amount == latest_balance - closing_fee (in this case 10_000)?

Yes, I thought about it but didn't want to hardcode the `10_000`. This was the best I could come up with 🙃

Reply:
I still think the assertion should be more like what I was suggesting. It should not matter if the fee is a global constant or an output of the protocol itself.

Reply:
Done